### PR TITLE
feat(`cheatcodes`): add `keyExists` cheatcode

### LINKED ISSUE #5369: vm.parseJson does not revert if key does not exist
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

_No response_

### What command(s) is the bug in?

_No response_

### Operating System

None

### Describe the bug

Right now `vm.parseJson('{"a": 1}', '.b')` will be executed successfully while `vm.parseJsonUint('{"a": 1}, '.b')` reverts with `"No matching value or array found for key .b"`. I think that `parseJson` should revert too if non-existent key is passed into it

### PR DESCRIPTION
**Pull request by Evalir**
## Motivation

Closes #5369 , supersedes https://github.com/foundry-rs/forge-std/pull/226 as a native impl of the cheatcode.

Instead of reverting if a key does not exist, a more correct behavior should be exposing a helper to validate if a key exists or not.

## Solution

Implements a `keyExists(string calldata json, string calldata key) external view returns (bool)` cheatcode, which can validate if a certain key exists or not.

cc @mds1

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
lgtm

File: evm/src/trace/utils.rs [hunk 1]
```diff
@@ -55,16 +55,17 @@ pub(crate) fn decode_cheatcode_inputs(
             Some(decoded.iter().map(format_token).collect())
         }
         "deriveKey" => Some(vec!["<pk>".to_string()]),
-        "parseJson" | "writeJson" => {
+        "parseJson" | "writeJson" | "keyExists" => {
```
Comment (mds1) [inline evm/src/trace/utils.rs:58]: Unrelated to this PR, but we should probably add all the `parseJson*` and `serialize*` methods too?

Comment (Evalir) [reply evm/src/trace/utils.rs:58]: yep highly agree, this can be done in another smol pr

File: abi/abi/HEVM.sol [hunk 2]
```diff
@@ -208,6 +208,7 @@ serializeString(string,string,string)(string)
 serializeString(string,string,string[])(string)
 serializeBytes(string,string,bytes)(string)
 serializeBytes(string,string,bytes[])(string)
+keyExists(string,string)(bool)
```
Comment (mds1) [inline abi/abi/HEVM.sol:211]: Do we like this name, or should it be `jsonKeyExists`? Thinking ahead to support for https://github.com/foundry-rs/foundry/issues/4808 where we'd have cheats to parse TOML also. TOML can always be converted to JSON, so I think the options are:
- Namespace with `jsonKeyExists` and `tomlKeyExists`
- Keep a single `keyExists` method and forge detects if the string is JSON or TOML and parses it accordingly

The latter feels like nicer UX so the user doesn't have to worry about the file format of the data

Comment (Evalir) [reply abi/abi/HEVM.sol:211]: Yep, I was going to rename to `jsonKeyExists` to keep namespaces separate, but actually I think `keyExists` with automatic detection leads to a nicer UX, and it shouldn't be too hard of an additional lift to implement.

**Pull request merged**